lib/Mojolicious/Plugin/MojoDocs.pm

```diff
@@ -29,15 +29,6 @@ sub _html {
   $_->{href} =~ s!^https://metacpan\.org/pod/!$perldoc! and $_->{href} =~ s!::!/!gi
     for $dom->find('a[href]')->map('attr')->each;
 
-  # Rewrite code blocks for syntax highlighting and correct indentation
-  for my $e ($dom->find('pre > code')->each) {
-    next if (my $str = $e->content) =~ /^\s*(?:\$|Usage:)\s+/m;
-    next unless $str =~ /[\$\@\%]\w|-&gt;\w|^use\s+\w/m;
```

Comment:
Where is this logic now?

Reply:
It is gone now because it only made sense for prettifyjs, while highlightjs already matches any code block in the DOM that matches `'pre > code'`.

Reply:
But these two lines with `next` seems to skip certain "code" tags..?

Reply:
Yep, yet all that after the `next` is simply adding a `prettyprint` CSS class to those remaining after the filters, which highlightjs doesn't need.

I _did_ think about retaining this filter for maybe adding explicit `language-perl language-mojolicious` instead, but I found that highlightjs just DWIMs enough that I did not need to over-think it.

Reply:
The skipping of some code blocks was very much intentional.

Reply:
If so maybe we can restore it and instead work to _disable_ highlighting for certain code blocks, adding either `plaintext` or `nohighlight` CSS classes to e.g. diagrams.

Reply:
I restored this block somewhat and changed its logic:

- start with `nohighlight` code blocks to preserve diagrams and listings
- remove `nohighlight` for blocks that we want highlighted

Reply:
This shouldn't overwrite the class attribute unconditionally, it should append nohighlight only if the logic determined it should.

Reply:
I thought of that first (see https://github.com/mojolicious/mojolicious.org/pull/22#discussion_r514328250) but when trying that, I found it was harder to reliably filter what should be _not_ highlighted, or worse, get highlighted in the _wrong_ way (e.g. diagrams getting highlighted as YAML!)

Since we're limited to rendering code blocks out of the Mojolicious distribution only, I thought that starting with `nohighlight` unconditionally should be OK for this case; let me know if there's another case that blocks this.

Reply:
That doesn't restrict the logic that can be used, you are still determining ultimately whether or not to add the class, it should just be reworked to only append it at the end.

Reply:
for example, you could invert the `next` conditions and then append nohighlight the same way prettyprint was before.

Reply:
There's a subtle difference: I'm determining whether or not to _keep_ `nohighlight` or delete it, rather than simply adding it.

Do you mean we could do something like this instead?

```perl
next if $class_should_be_highlighted;
my $class = $e->attr->{class};
$e->attr->{class} = defined $class ? "$class nohighlight" : 'nohighlight';
```

Reply:
Correct.

Reply:
Cool, I'll get around to applying and testing that in a bit :muscle:

Reply:
Ok, so given this:

```perl
    next unless (my $str = $e->content) =~ /^\s*(?:\$|Usage:)\s+/m;
    next if $str =~ /[\$\@\%]\w|-&gt;\w|^use\s+\w|^#|^&lt;|^\[/m;
    my $class = $e->attr->{class};
    $e->attr->{class} = defined $class ? "$class nohighlight" : 'nohighlight';
```
It renders the Perl code blocks fine, but breaks diagrams (e.g. in Cookbook:)

![image](https://user-images.githubusercontent.com/110625/97660467-a16e7580-1aac-11eb-8b13-4e775fdee223.png)

and also sample `Usage:` listings (e.g. in `Mojolicious::Command::daemon`:)

![image](https://user-images.githubusercontent.com/110625/97660513-c531bb80-1aac-11eb-83ac-ad9d1e16cdde.png)

I _think_ a compromise might be something like this instead (With `$e->attr->{class} = $class // ''` to avoid rendering `<pre><code class>...`:)

```perl
    my $class = $e->attr->{class};
    $e->attr->{class} = defined $class ? "$class nohighlight" : 'nohighlight';
    next if (my $str = $e->content) =~ /^\s*(?:\$|Usage:)\s+/m;
    next unless $str =~ /[\$\@\%]\w|-&gt;\w|^use\s+\w|^#|^&lt;|^\[/m;
    $e->attr->{class} = $class // '';
```

What do you think?

Reply:
Your first attempt fails because the two `next` conditions is effectively an `or` compound condition, and so inverted needs to be an `and` e.g.

```perl
next if (my $str = $e->content) !~ /^\s*(?:\$|Usage:)\s+/m
  and $str =~ /[\$\@\%]\w|-&gt;\w|^use\s+\w|^#|^&lt;|^\[/m;
```

Reply:
Thanks, yeah I missed inverting the first condition _and_ the regex match.  It now work with this, thanks!

```perl
    next if $e->content !~ /^\s*(?:\$|Usage:)\s+/m
      and $e->content =~ /[\$\@\%]\w|-&gt;\w|^use\s+\w|^#|^&lt;|^\[/m;
    my $class = $e->attr->{class};
    $e->attr->{class} = defined $class ? "$class nohighlight" : 'nohighlight';
```